lib/logflare_web/controllers/api/team_controller.ex

```diff
@@ -0,0 +1,51 @@
+defmodule LogflareWeb.Api.TeamController do
+  use LogflareWeb, :controller
+  alias Logflare.Teams
+  action_fallback LogflareWeb.Api.FallbackController
+
+  def index(%{assigns: %{user: user}} = conn, _) do
+    teams = Teams.list_for_user(user)
+    json(conn, teams)
+  end
+
+  def show(%{assigns: %{user: user}} = conn, %{"token" => token}) do
+    with team when not is_nil(team) <- Teams.get(user, token),
+         team <- Teams.preload_fields(team, [:user, :team_users]) do
+      json(conn, team)
+    end
+  end
+
+  def create(%{assigns: %{user: user}} = conn, params) do
+    with params <- params_keys_string_to_atom(params),
+         {:ok, team} <- Teams.create_team(user, params),
+         team <- Teams.preload_fields(team, [:user, :team_users]) do
+      conn
+      |> put_status(201)
+      |> json(team)
+    end
+  end
+
+  def update(%{assigns: %{user: user}} = conn, %{"token" => token} = params) do
+    with params <- params_keys_string_to_atom(params),
+         team when not is_nil(team) <- Teams.get_team_by(token: token, user_id: user.id),
+         {:ok, team} <- Teams.update_team(team, params),
+         team <- Teams.preload_fields(team, [:user, :team_users]) do
+      conn
+      |> put_status(204)
+      |> json(team)
+    end
+  end
+
+  def delete(%{assigns: %{user: user}} = conn, %{"token" => token}) do
+    with team when not is_nil(team) <- Teams.get_team_by(token: token, user_id: user.id),
+         {:ok, _} <- Teams.delete_team(team) do
+      conn
+      |> Plug.Conn.send_resp(204, [])
+      |> Plug.Conn.halt()
+    end
+  end
+
+  defp params_keys_string_to_atom(params) do
+    params |> Enum.map(fn {k, v} -> {String.to_existing_atom(k), v} end) |> Enum.into(%{})
```

Comment:
why is this needed? Not very safe. ecto changeset should be used for casting.

Reply:
wasn't working for some reason 😞 unsure why

Reply:
stupid me it was something else for sure because it works... will fix the code